client-v2/src/shared/components/input/InputImage.tsx

```diff
@@ -110,17 +114,112 @@ interface ComponentState {
   imageSrc: string;
 }
 
+const dragImage = new Image();
+dragImage.src = imageDragIcon;
+
 class InputImage extends React.Component<ComponentProps, ComponentState> {
   public state = {
     isHovering: false,
     modalOpen: false,
     imageSrc: ''
   };
 
-  public handleDrop = (e: React.DragEvent<HTMLDivElement>) => {
+  public render() {
+    const { small, input, gridUrl } = this.props;
+    const gridSearchUrl = `${gridUrl}?cropType=landscape`;
+    const hasImage = !!input.value && !!input.value.thumb;
```

Comment:
Small thing, but why the double negative? `!!`

Reply:
Good question! The double negative coerces the expression that follows it into a boolean type. I think it's fair to say that this is idiomatic in Javascript. In this case it ensures that `hasImage` is of a boolean type. Because of short-circuit evaluation, if the double negatives weren't there, 
- if input.value was falsey, hasImage would be the value of input.value, which could be _anything falsey_ (`undefined`, `null`, `false`, `NaN`, maybe I've missed a few!)
- if input.value was truthy but input.value.thumb was not, hasImage would be the value of input.value.
- if input.value and input.value.thumb were both truthy, hasImage would be the value of input.value.thumb.

Reply:
thanks for explanation!